extensions/hidemyemail/src/list-emails.tsx

```diff
@@ -0,0 +1,268 @@
+        <List.Dropdown.Item title="Active" value={Status.ACTIVE} />
+        <List.Dropdown.Item title="Inactive" value={Status.INACTIVE} />
+      </List.Dropdown.Section>
+    </List.Dropdown>
+  );
+}
+
+export default function Command() {
+  const [service, setService] = useState<iCloudService | null>(null);
+  const [emails, setEmails] = useState<Array<HideMyEmail> | null>(null);
+  const [status, setStatus] = useState<Status>(Status.ANY);
+  const { sortByCreationDate } = getPreferenceValues<Preferences.ListEmails>();
+
+  useEffect(() => {
+    (async () => {
+      await updateAddressList();
+    })();
+  }, [service]);
+
+  if (!service) {
+    return <Login onLogin={(iService: iCloudService) => setService(iService)} />;
+  }
+
+  if (emails === null) {
+    return <List isLoading={true} />;
+  }
+
+  async function updateAddressList() {
+    if (service) {
+      const data = await service.hideMyEmail.getAllAdresses();
+      const hmeEmails = data?.["hmeEmails"];
+      if (hmeEmails) {
+        if (sortByCreationDate)
+          hmeEmails.sort((hme1: HideMyEmail, hme2: HideMyEmail) => hme2.createTimestamp - hme1.createTimestamp);
+
+        setEmails(hmeEmails);
+      }
+    }
+  }
+
+  async function toggleActive(email: HideMyEmail) {
+    const toast = await showToast({ style: Toast.Style.Animated, title: "Updating..." });
+
+    try {
+      await service?.hideMyEmail.toggleActive(email.anonymousId, email.isActive ? "deactivate" : "reactivate");
+      await updateAddressList();
+      toast.style = Toast.Style.Success;
+      toast.title = "Updated";
+    } catch (error) {
+      toast.style = Toast.Style.Failure;
+      toast.title = "Update failed";
+      toast.message = (error as { message: string }).message;
+    }
+  }
+
+  async function updateMetaData(email: HideMyEmail, newMetaData: MetaData) {
+    const toast = await showToast({ style: Toast.Style.Animated, title: "Updating..." });
+    try {
+      await service?.hideMyEmail.updateMetaData(email.anonymousId, newMetaData);
+      await updateAddressList();
+      toast.style = Toast.Style.Success;
+      toast.title = "Label updated";
+    } catch (error) {
+      toast.style = Toast.Style.Failure;
+      toast.title = "Update failed";
+      toast.message = (error as { message: string }).message;
+    }
+  }
+
+  async function remove(email: HideMyEmail) {
+    const toast = await showToast({ style: Toast.Style.Animated, title: "Updating..." });
+    try {
+      if (email.isActive) {
+        await service?.hideMyEmail.toggleActive(email.anonymousId, "deactivate");
+      }
+      await service?.hideMyEmail.deleteAddress(email.anonymousId);
+      await updateAddressList();
+      toast.style = Toast.Style.Success;
+      toast.title = "Address deleted";
+    } catch (error) {
+      toast.style = Toast.Style.Failure;
+      toast.title = "Update failed";
+      toast.message = (error as { message: string }).message;
+    }
+  }
```

Comment:
```suggestion
  async function remove(email: HideMyEmail) {
    await confirmAlert({
      title: "Are you sure?",
      message: "This will delete the email address.",
      primaryAction: {
        title: "Delete",
        style: Alert.ActionStyle.Destructive,
        onAction: async () => {
          const toast = await showToast({ style: Toast.Style.Animated, title: "Updating..." });
          try {
            if (email.isActive) {
              await service?.hideMyEmail.toggleActive(email.anonymousId, "deactivate");
            }
            await service?.hideMyEmail.deleteAddress(email.anonymousId);
            await updateAddressList();
            toast.style = Toast.Style.Success;
            toast.title = "Address deleted";
          } catch (error) {
            toast.style = Toast.Style.Failure;
            toast.title = "Update failed";
            toast.message = (error as { message: string }).message;
          }
        },
      },
      dismissAction: {
        title: "Cancel",
      },
      rememberUserChoice: true,
    });
  }
```